hashintel/hash
# Implement read-only UI in blocks

## 🌟 What is the purpose of this PR?



This PR updates all blocks in the repo to render `read-only` UI when they receive `readonly` prop

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1202621375435414/1202643260631376) _(internal)_


## 🔍 What does this change?




- Disables/hides triggers that call updateEntity in each block in read-only mode. Blocks can't call updateEntity in read-only mode but can fetch latest entity data
- Updates table block to call `aggregateEntities` for filter and sort operations in read-only mode to prevent the operations from being persisted


## 🐾 Next steps


- Update blocks in BP hub

## 🛡 What tests cover this?



- New playwright tests

## ❓ How to test this?



1.  Load a local block (read-only version of blocks isn't in Block Hub yet) in a page
1.  Switch the page to readonly mode (add `?readonly` to the url)
1.  Confirm changes can't be made to the block

## 📹 Demo



https://user-images.githubusercontent.com/16928688/183587537-9382049a-da47-4eec-9b87-c749106947fc.mov

Editable version on the left, Read-only version on the right



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202643260631376
